components/prism-typoscript.js

```diff
@@ -0,0 +1,79 @@
+(function(Prism) {
+
+	var keywords = /\b(?:ACT|ACTIFSUB|CARRAY|CASE|CLEARGIF|COA|COA_INT|CONSTANTS|CONTENT|CUR|EDITPANEL|EFFECT|EXT|FILE|FLUIDTEMPLATE|FORM|FRAME|FRAMESET|GIFBUILDER|GMENU|GMENU_FOLDOUT|GMENU_LAYERS|GP|HMENU|HRULER|HTML|IENV|IFSUB|IMAGE|IMGMENU|IMGMENUITEM|IMGTEXT|IMG_RESOURCE|INCLUDE_TYPOSCRIPT|JSMENU|JSMENUITEM|LLL|LOAD_REGISTER|NO|PAGE|RECORDS|RESTORE_REGISTER|TEMPLATE|TEXT|TMENU|TMENUITEM|TMENU_LAYERS|USER|USER_INT|_GIFBUILDER|global|globalString|globalVar)\b/;
+
+	Prism.languages.typoscript = {
+		'comment': [
+			{
+				// multiline comments /* */
+				pattern: /[^\/]\/\*[^.][\s\S]*?\*\//,
+			},
+			{
+				// double-slash comments - ignored when backslashes or colon is found in front
+				// also ignored whenever directly after an equal-sign, because it would probably be an url without protocol
+				pattern: /(^|[^\\:= \t]|(?:^|[^= \t])[ \t]+)\/\/.*/,
+				lookbehind: true,
+				greedy: true
+			},
+			{
+				// hash comments - ignored when leading quote is found for hex colors in strings
+				pattern: /(^|[^"'])#.*/,
+				lookbehind: true,
+				greedy: true
+			}
+		],
+		'function': [
+			{
+				// old include style
+				pattern: /<INCLUDE_TYPOSCRIPT:\s*source\s*=\s*['"].*['"]\s*>/,
+				inside: {
+					'string': {
+						pattern: /['"].*['"]/,
+						inside: {
+							'keyword': keywords,
+						},
+					},
+					'keyword': {
+						pattern: /INCLUDE_TYPOSCRIPT/,
+					},
+				},
+			},
+			{
+				// new include style
+				pattern: /@import\s*['"].*['"]/,
+				inside: {
+					'string': /['"].*['"]/,
+				},
+			}
+		],
+		'string': {
+			pattern: /^([^=]*=[< ]?)(?:(?!]\n).)*/,
+			lookbehind: true,
+			inside: {
+				'function': /{\$.*}/, // constants include
+				'keyword': keywords,
+				'number': /^[0-9]+$/,
+				'punctuation': /[,|:]/,
+			}
+		},
+		'keyword': keywords,
+		'number': {
+			// special highlighting for indexes of arrays in tags
+			pattern: /[0-9]+\s*[.{=]/,
+			inside: {
+				'operator': /[.{=]/,
+			}
+		},
+		'tag': {
+			pattern: /\s*\.?[A-z0-9-_]+\.?\s*/,
```

Comment:
1. `A-z`: That's probably not right, is it? This will match all characters from `A` (`\x41`) to `z` (`\x7a`) including characters like these: `@_[]`.
2. Please remove the leading and trailing `\s*`.

```suggestion
			pattern: /\.?[\w-]+\.?/,
```

Reply:
I only found `\` as other character used in string, so i went with `/\.?[\w-\\]+\.?/`.